src/frontend/src/binder/expr/mod.rs

```diff
@@ -478,7 +485,63 @@ impl Binder {
             .map(|expr| self.bind_expr_inner(*expr))
             .transpose()?;
 
+        let mut constant_lookup_inputs = Vec::new();
```

Comment:
The function `bind_case` mixes the optimization and the normal situation and makes the code not straightforward. I think we could do a check first to decide whether we could perform this optimization and then separate the construction of logic `ExprType::ConstantLookup` and `ExprType::Case` into different functions.